executor/aggfuncs/func_group_concat.go

```diff
@@ -491,28 +515,32 @@ func (e *groupConcatDistinctOrder) UpdatePartialResult(sctx sessionctx.Context,
 			continue
 		}
 		p.valSet.Insert(joinedVal)
+		memDelta += int64(len(joinedVal))
 		sortRow := sortRow{
 			buffer:  buffer,
 			byItems: make([]*types.Datum, 0, len(e.byItems)),
 		}
 		for _, byItem := range e.byItems {
 			d, err := byItem.Expr.Eval(row)
 			if err != nil {
-				return 0, err
+				return memDelta, err
 			}
 			sortRow.byItems = append(sortRow.byItems, d.Clone())
```

Comment:
`sortRow.byItems` had append a lot of Datums. should we need to evaluate the memory usage of it also? @b41sh @William0423

Reply:
I agree, we should add the size of row

Reply:
ok

Reply:
Should we evaluate the memDelta according to the Datum's type?

Reply:
Ok, I will add memory usage of Datums.

Reply:
@William0423 @b41sh
I think we need to calculate the `memDelta` in the `tryToAdd` which has `heap.push`, `heap.pop` and `buffer.Truncate` operations that will affect the memory usage. What your opinions?

Reply:
@mmyj 
I think `memDelta - the memory usage of heap.Pop(h) ` shall be calculated. `currSize` has calculated memory changes for   `buffer.Truncate`. `heap.Push(h, row)` seems have no effect on `memDelta`.

Reply:
I agree, IMO we could write the memdelta calculated logic in the tryToAdd function.

Reply:
yes, we can return `memDelta` in the `tryToAdd` function.

Reply:
ok